pyuvm/s24_pyuvm_reg_includes.py

```diff
@@ -1,121 +1,178 @@
-                # Return the function unchanged, not decorated. if use_pyvsc is not enabled
+            if (use_pyvsc is True):
+                raise UVMNotImplemented()
+            else:
+                # Return the function unchanged, not decorated. if use_pyvsc
+                # is not enabled
                 return self.cls
     return enable_rand
 
+
 '''
-enable_pyvsc                            = False ## Global to be set in case we wanna use the VSC package
-enable_auto_predict                     = True  ## Global to be set in case we wanna use the auto prediction
-enable_throw_error_response_on_read     = False ## Global enable bit for error response in case of NO-EFFECT action based on the access type
-enable_throw_error_response_on_write    = False ## Global enable bit for error response in case of NO-EFFECT action based on the access type
-disable_code_interruption_assert        = False ## Global assert disable this is used in order to avoid code interruption by just checking for the error list
+enable_pyvsc: False
+Global to be set in case we wanna use the VSC package
+
+enable_auto_predict: True
+Global to be set in case we wanna use the auto prediction
+
+enable_throw_error_response_on_read: False
+Global enable bit for error response
+in case of NO-EFFECT action based on the access type
+
+enable_throw_error_response_on_write: False
+Global enable bit for error response
+in case of NO-EFFECT action based on the access type
+
+disable_code_interruption_assert: False
+Global assert disable this is used in
+order to avoid code interruption by just checking for the error list
 '''
-enable_pyvsc                            = False 
-enable_auto_predict                     = True
-enable_throw_error_response_on_read     = False 
-enable_throw_error_response_on_write    = False 
-disable_code_interruption_assert        = False 
-
-def error_out(header,message):
-    ''' 
+enable_pyvsc = False
+enable_auto_predict = True
+enable_throw_error_response_on_read = False
+enable_throw_error_response_on_write = False
+disable_code_interruption_assert = False
+disable_code_interruption_fatal = False
+
+
+#
+def uvm_error(header="", message=""):
+    '''
     Used to error out based on header and message
     '''
-    assert(disable_code_interruption_assert,header+message)
+    raise UVMError(f"{header+message}")
+
 
+#
+def uvm_fatal(header="", message=""):
+    '''
+    Used to fatal out based on header and message
+    '''
+    if (disable_code_interruption_fatal is False):
+        raise UVMFatalError(f"{header+message}")
+
+
+#
+def uvm_not_implemeneted(header="", message=""):
+    '''
+    Used to fatal out based on header and message
+    '''
+    if (disable_code_interruption_fatal is False):
+        raise UVMNotImplemented(f"{header+message}")
+
+
+#
+def error_out(header, message):
+    '''
+    Used to error out based on header and message
+    '''
+    assert disable_code_interruption_assert, header + message
+
+
+#
```

Comment:
implemented `uvm_not_implemeneted`, `uvm_fatal` and `uvm_error` as functions will result handy in the future. This is not the right place to put them but for the time being would be helpful